### DISCUSSION TITLE
Question about Dietpi-Backup

### DISCUSSION THREAD

**Discussion by pedrom34** [MichaIng/DietPi#5914] [Q&A] [answered]
Hey,

I have dietpi_userdata on a usbkey. I want to backup dietpi on another usb key, and I cannot figure out how to do it with filters...
I've read this https://github.com/MichaIng/DietPi/discussions/5836
and the related help: https://dietpi.com/docs/dietpi_tools/#dietpi-backup-backuprestore

Here's my /mnt list:
Folder | Comment
---|---
/mnt/samba/                               | Empty
/mnt/nas/                                 | NFS Mount
/mnt/containers/                          | Symlink
/mnt/depot/                               | Symlink
/mnt/dietpi_userdata/                     | Symlink
/mnt/docs/                                | Symlink
/mnt/downloads/                           | Symlink
/mnt/films/                               | Symlink
/mnt/music/                               | Symlink
/mnt/series/                              | Symlink
/mnt/syncthing/                           | Symlink
/mnt/b002350b-019e-40e4-9177-1e678666e503/| USB Key for backup
/mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/| USB Key for dietpi_userdata

So, I want the dietpi_userdata to be included in the backup made on /mnt/b002350b-019e-40e4-9177-1e678666e503/ but I don't want the nas to be copied...
Default filter doesn't include it.
I tried:
```
+ /mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/dietpi_userdata/
- /mnt/*
- /media/
```

I tried:
```
+ /mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/dietpi_userdata/*
- /mnt/
```

Can you help me?

**Comment by MichaIng** [comment, marked as answer]
Try this:
```
+ /mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/
- /mnt/*
- /media/
```
This is the relevant info from the default filter file:
```
# Excluded directories are not processed recursively, so contained items cannot be included.
```
So with your filter, `/mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/dietpi_userdata` is not processed, because `/mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1` is excluded via `- /mnt/*` and `- /mnt/` the same way. If parent directories match an exclude rule, preceding include rules need to include those to be processed. With the above example, all content of the USB drive is backed up. If you really only want include the userdata directory, the following would do:
```
+ /mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/dietpi_userdata/
- /mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/*
+ /mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/
- /mnt/*
- /media/
```
Somehow complicated, I know, but once you understand the logic, it somehow makes sense. The list is always processed from top to bottom on every processed file, applying the first rule which matches. So in above case:
- The directory `/mnt/` does not match any rule, as `/mnt/*` matches only all contained items, not the directory itself.
- `/mnt/containers/` etc match line 4, hence are excluded, like all items contained in `/mnt/`, aside of:
- `/mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/` matches line 3, hence is included, and line 4 is then not applied anymore.
- `/mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/something` matches line 2, hence is excluded like every other item in `/mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/`, aside of:
- `/mnt/c7a8c22e-6b5b-4425-9731-23379c3655b1/dietpi_userdata/` is included, as it matches line 1 and hence line 2 is not applied anymore.

An info that is not mentioned in the default filter file:
- If a directory is included, all it's content is included recursively, if there is no preceding exclude rule on contained items. So this is consistent to directory exclude rules, which as well apply to any contained items, as long as there is not preceding include rule on contained items.

**Reply by pedrom34** [reply]
Thank you a lot Micha!
Well, yes, it's a bit complex to understand 😅

**Comment by StephanStS** [comment]
@MichaIng : I propose to close this discussion.
With this commit (https://github.com/MichaIng/DietPi-Docs/commit/3d703da99803a0d43aba0dbb6082e8a27b2b1bf4) we added further information about the filtering.
Also in our blog there is a bit more explanation: https://dietpi.com/blog/?p=1983#BackupSettings
